### PR TITLE
Fix the bug of the metadata of the multi-language site.

### PR DESCRIPTION
**Pull request by FledgeXu**
Fix: #259

### CODE REVIEW

**Comment by kelson42** [comment]
@rgaudin We should check the Name metadata as well

**Comment by FledgeXu** [comment]
@kelson42 So far, the value  the `Name` metadata is the file name.

**Comment by kelson42** [comment]
@FledgeXu … and it’s wrong. For the SE about French language for example:
* filename should be `french.stackexchange.com_mul_all_2023-03.zim`
* Name metadata should be `french.stackexchange.com_mul_all`

Can you please fix these problems as well please?

File: src/sotoki/constants.py [hunk 1]
```diff
@@ -70,10 +70,10 @@ def lang_for_domain(domain):
                 lang = get_language_details(so_code)
                 if not lang["iso-639-1"] or not lang["iso-639-3"]:
                     raise NotFound("Might be an abbreviation")
-                return lang["iso-639-1"], lang["iso-639-3"]
+                return lang["iso-639-1"], lang["iso-639-3"], True
```
Comment (kelson42) [inline src/sotoki/constants.py:73]: It’s not clear what this last parameter should, please document clearly the whole function

File: src/sotoki/constants.py [hunk 2]
```diff
@@ -189,9 +189,20 @@ def any_restriction(self):
     def __post_init__(self):
         self.dump_domain = self.domain  # dumps are named after unfixed domains
         self.domain = FIXED_DOMAINS.get(self.domain, self.domain)
-        self.iso_lang_1, self.iso_lang_3 = lang_for_domain(self.domain)
+        self.iso_lang_1, self.iso_lang_3, is_multi_language = lang_for_domain(
```
Comment (kelson42) [inline src/sotoki/constants.py:192]: Here you should probably change the whole prototype to return lists instead of adding an additional parameter.

**Comment by FledgeXu** [comment]
@kelson42 I just check the `Name` metadata, it looks right?
image

**Comment by FledgeXu** [comment]
I just update the code for the clear logic.

File: src/sotoki/constants.py [hunk 3]
```diff
@@ -48,6 +48,13 @@
 
 
 def lang_for_domain(domain):
+    """
```
Comment (rgaudin) [inline src/sotoki/constants.py:51]: Doesn't respect our code formatting (black): single summary on same line then a blank one then the rest of the docstring

File: src/sotoki/constants.py [hunk 4]
```diff
@@ -189,9 +196,19 @@ def any_restriction(self):
...
         self.domain = FIXED_DOMAINS.get(self.domain, self.domain)
-        self.iso_lang_1, self.iso_lang_3 = lang_for_domain(self.domain)
+        iso_lang_1_codes, iso_lang_3_codes = lang_for_domain(self.domain)
+        iso_lang_1_primary_code = iso_lang_1_codes[0]
         variant = "nopic" if self.without_images else "all"
-        self.name = self.name or f"{self.domain}_{self.iso_lang_1}_{variant}"
+        if not self.name:
+            # if domain is a multi-language, add `multi` to file name.
+            if len(iso_lang_1_codes) > 1:
+                self.name = f"{self.domain}_multi_{iso_lang_1_primary_code}_{variant}"
```
Comment (rgaudin) [inline src/sotoki/constants.py:205]: The code for use in `Name` and in filename is `mul` not `multi`

**Review by rgaudin — COMMENTED** [review]
Thank you ; while reviewing your code, I realized there was an issue with your previous PR and ended up fixing both at once. Sorry about that.
See 2671642 for the details.

I am reviewing this still so information is shared.

**Comment by FledgeXu** [comment]
@rgaudin Thanks for your reviews! Although it's not merged, I still learnt a lot.

**Pull request closed without merging**